src/telegram/error.py

```diff
@@ -208,21 +213,43 @@ class RetryAfter(TelegramError):
        :attr:`retry_after` is now an integer to comply with the Bot API.
 
     Args:
-        retry_after (:obj:`int`): Time in seconds, after which the bot can retry the request.
+        retry_after (:obj:`int` | :class:`datetime.timedelta`): Time in seconds, after which the
+            bot can retry the request.
+
+            .. versionchanged:: NEXT.VERSION
+                |time-period-input|
 
     Attributes:
-        retry_after (:obj:`int`): Time in seconds, after which the bot can retry the request.
+        retry_after (:obj:`int` | :class:`datetime.timedelta`): Time in seconds, after which the
+            bot can retry the request.
+
+            .. deprecated:: NEXT.VERSION
+                |time-period-int-deprecated|
 
     """
 
-    __slots__ = ("retry_after",)
+    __slots__ = ("_retry_after",)
+
+    def __init__(self, retry_after: TimePeriod):
+        self._retry_after: dtm.timedelta = parse_period_arg(  # type: ignore[assignment]
+            retry_after
+        )
+
+        if isinstance(self.retry_after, int):
+            super().__init__(f"Flood control exceeded. Retry in {self.retry_after} seconds")
+        else:
+            super().__init__(f"Flood control exceeded. Retry in {self.retry_after!s}")
 
-    def __init__(self, retry_after: int):
-        super().__init__(f"Flood control exceeded. Retry in {retry_after} seconds")
-        self.retry_after: int = retry_after
+    @property
+    def retry_after(self) -> Union[int, dtm.timedelta]:  # noqa: D102
+        # Diableing D102 because docstring for `retry_after` is present at the class's level
+        return get_timedelta_value(  # type: ignore[return-value]
+            self._retry_after, attribute="retry_after"
+        )
 
     def __reduce__(self) -> tuple[type, tuple[float]]:  # type: ignore[override]
-        return self.__class__, (self.retry_after,)
+        # Until support for `int` time periods is lifted, leave pickle behaviour the same
```

Comment:
```suggestion
        # Until support for `int` time periods is lifted, leave pickle behaviour the same
        # tag: deprecated: NEXT.VERSION
```

just for finding it easier : ) nice catch!